kivymd/uix/button/button.kv

```diff
@@ -32,15 +32,26 @@
 
 <ButtonContentsText>
     lbl_txt: lbl_txt
-    width: max(root._min_width, lbl_txt.texture_size[0] + dp(24))
-    size_hint_min_x: max(root._min_width, lbl_txt.texture_size[0] + dp(24))
-    height: dp(20) + lbl_txt.texture_size[1]
+    width:
+        max(root._min_width, \
+        root.padding[0] + lbl_txt.texture_size[0] + root.padding[2])
+    size_hint_min_x:
+        max(root._min_width, \
+        root.padding[0] + lbl_txt.texture_size[0] + root.padding[2])
+    height:
+        max(root._min_height, \
+        root.padding[1] + lbl_txt.texture_size[1] + root.padding[3])
+    size_hint_min_y:
+        max(root._min_height, \
+        root.padding[1] + lbl_txt.texture_size[1] + root.padding[3])
 
     MDLabel:
         id: lbl_txt
         text: root.text
         font_size: root.font_size
         font_style: root.font_style
+        halign: 'center'
+        valign: 'middle'
```

Comment:
This will not allow changing the position of the text.

Reply:
You can't do that anyway? (well not easily)
This only affects multi-line text. If you want to move the whole text label around within the button, the existing halign and valign on the _button_, not the label, would be the things to do that with (although I don't know if they actually work). It would, I think, also not be in accordance with the MD spec.

Reply:
> You can't do that anyway? (well not easily) This only affects multi-line text. If you want to move the whole text label around within the button, the existing halign and valign on the _button_, not the label, would be the things to do that with (although I don't know if they actually work). It would, I think, also not be in accordance with the MD spec.

It worked. But you removed this feature in your previous changes.

Reply:
See the last set of screenshots which show how halign on the _label_, not the _button_, affect the positioning of multi-line text.

Reply:
> It worked. But you removed this feature in your previous changes.

halign and valign on the _button_ still work:

![image](https://user-images.githubusercontent.com/4930097/146792009-19a5b198-1c60-472c-858b-b95bdc08bef3.png)

https://pastebin.com/pE020Rbu